### PR TITLE
feat: added loading spinner

### PR DESCRIPTION
**Pull request by xpavlic**
When adding or removing a token on manage tokens page,
content is hidden, loading spinner is displayed and
buttons are disabled. Credential_loading function
checks every second if the change already took effect
with 1 minute loading timeout.

Merging is possible after merge of #44 .

### CODE REVIEW

File: app/static/js/satosa_webauthn.js [hunk 1]
```diff
@@ -246,11 +247,36 @@ async function registerClickHandler(form, event) {
...
+
+function toggleLoading() {
+    toggleElementDisplay(document.getElementById('content-block'));
+    toggleElementDisplay(document.getElementById('loading-block'));
+    const logout_button = document.getElementById('logout-button');
+    if (logout_button.classList.contains("disabled")) {
+        logout_button.classList.replace("disabled", "active");
+    } else {
+        logout_button.classList.add("disabled");
+    }
```
Comment (melanger) [inline app/static/js/satosa_webauthn.js:272]: If initially the logout button has exactly 1 of `disabled` or `active`:
```
logout_button.classList.toggle("disabled");
logout_button.classList.toggle("active");
```

File: app/static/js/satosa_webauthn.js [hunk 2]
```diff
@@ -246,11 +247,36 @@ async function registerClickHandler(form, event) {
...
+}
+
+function toggleElementDisplay(element) {
+    if (element.classList.contains("d-none")) {
+        element.classList.replace("d-none", "d-block");
+    } else if (element.className.match("d-block")) {
+        element.classList.replace("d-block", "d-none");
     }
```
Comment (melanger) [inline app/static/js/satosa_webauthn.js:280]: If initally the `element` has exactly one of `d-none` or `d-block`:
```
element.classList.toggle("d-none");
element.classList.toggle("d-block");
```

**Review by melanger — APPROVED** [review]
@xpavlic Rebased on the current main and slightly changed to keep compatibility with MUNI framework.

It would be nice to hide a smaller portion of the UI, instead of hiding everything and just showing the spinner, but it works nicely.

**Comment by xpavlic** [comment]
Register and delete tokens process now hide just connected conent, show loading spinner instead and disable register, delete and continue buttons.

File: app/templates/credentials.html [hunk 3]
```diff
@@ -28,24 +30,31 @@
...
+                        <span>{{ _('Turn off WebAuthn authentication') }}</span>
                     </button>
                 </p>
             </form>
         </div>
     {% endif %}
-    <table class="table table-dark table-striped align-middle mt-5">
+    <div id="delete-loading" class="d-none justify-content-center hide">
+        <div class="spinner-border mt-5 mb-3" role="status">
+            <span class="visually-hidden">Loading...</span>
```
Comment (melanger) [inline app/templates/credentials.html:54]: change `Loading...` to `{{ _('Loading…') }}` (i18n + ellipsis character) and add a translation

File: app/templates/credentials.html [hunk 4]
```diff
@@ -70,7 +79,13 @@
     </table>
     <div class="{{ style_classes['card'] }}">
         <h2 class="mb-4">{{ _('Register new token') }}</h2>
-        <form id="register-form" class="{{ style_classes['form'] }}" name="register" method="POST" action="{{ url_for('webauthn_begin_activate') }}">
+        <div id="register-loading" class="d-none justify-content-center hide">
+            <div class="spinner-border mt-5 mb-3" role="status">
+                <span class="visually-hidden">Loading...</span>
+            </div>
```
Comment (melanger) [inline app/templates/credentials.html:85]: include the same spinner as in `app/templates/credentials.html`

**Review by melanger — CHANGES_REQUESTED** [review]

**Comment by melanger** [comment]
Please add translations for all text that is shown to user (everything except log messages)

File: app/templates/credentials.html (hunk 4 above)
Comment (xpavlic) [reply app/templates/credentials.html:85]: I am not sure what do you mean. Both spinners in `app/templates/credentials.html` are same.

Comment (melanger) [reply app/templates/credentials.html:85]: That is the problem, the same source code is copied in two places. Please move it to a single file which is included in two locations.

File: app/templates/credentials.html [hunk 5]
```diff
@@ -28,24 +30,28 @@
...
                     </p>
                 </form>
             {% endif %}
-            <form action="{{ url_for('turn_on_auth') }}" class="{{ style_classes['form'] }} mt-3" id="turn_on_form">
+            <form action="{{ url_for('turn_on_auth') }}" id="turn_on_form"
+                  class="{{ style_classes['form'] }} mt-3">
                 <p class="{{ style_classes['button_wrap'] }}">
                     <button class="{{ style_classes['form_button'] }}" type="submit">
-                        <span>{{ _('Turn on WebAuthn authentication') }}</span>
+                        <span>{{ _('Turn off WebAuthn authentication') }}</span>
```
Comment (melanger) [inline app/templates/credentials.html:45]: this seems like a typo

**Review by melanger — APPROVED** [review]

**Pull request merged**